torchdiffeq/_impl/adjoint.py

```diff
@@ -182,22 +162,35 @@ def odeint_adjoint(func, y0, t, *, rtol=1e-7, atol=1e-9, method=None, options=No
         adjoint_atol = atol
     if adjoint_method is None:
         adjoint_method = method
+
     if adjoint_options is None:
         adjoint_options = {k: v for k, v in options.items() if k != "norm"} if options is not None else {}
+    else:
+        adjoint_options = adjoint_options.copy()
+
     if adjoint_params is None:
         adjoint_params = tuple(find_parameters(func))
     else:
         adjoint_params = tuple(adjoint_params)  # in case adjoint_params is a generator.
 
     # Filter params that don't require gradients.
+    for p in adjoint_params:
+        if not p.requires_grad:
+            # Issue a warning if a user-specified norm is specified.
+            if 'norm' in adjoint_options and adjoint_options['norm'] != 'seminorm':
+                warnings.warn("An adjoint parameter was passed without requiring gradient. For efficiency this will be "
+                              "excluded from the adjoint pass, and will not appear as a tensor in the adjoint norm.")
+
     adjoint_params = tuple(p for p in adjoint_params if p.requires_grad)
 
-    # Normalise to non-tupled input
+    # Normalise to non-tupled state.
     shapes, func, y0, t, rtol, atol, method, options, event_fn, decreasing_time = _check_inputs(func, y0, t, rtol, atol, method, options, event_fn, SOLVERS)
 
-    if "norm" in options and "norm" not in adjoint_options:
-        adjoint_shapes = [torch.Size(()), y0.shape, y0.shape] + [torch.Size([sum(param.numel() for param in adjoint_params)])]
-        adjoint_options["norm"] = _wrap_norm([_rms_norm, options["norm"], options["norm"]], adjoint_shapes)
+    # Avoid in-place modifying a user-specified dict.
+    adjoint_options = adjoint_options.copy()
```

Comment:
This is unnecessary as it has already been copied above.

Reply:
Ah, good catch.